### ISSUE TITLE
Configuration.__contains__ still does not work 

### ISSUE THREAD

**Issue by mdorier** [automl/ConfigSpace#383]
I can't reopen https://github.com/automl/ConfigSpace/issues/378, but the fix is actually a bug.
What you want here https://github.com/automl/ConfigSpace/blob/main/src/ConfigSpace/configuration.py#L180 is `return key in self._values`, because the `config_space` may have parameters that the `Configuration` does not have, due to conditions.

**Comment by eddiebergman** [comment]
Damn, sorry I did that too hastily. Looking deeper, that's also not a fully valid option as the `._values` is not populated until used (we instead store a vectorized array until values are actually needed from it). https://github.com/automl/ConfigSpace/blob/ef48f6daaec60145b8323b8e84132d12efa895ba/src/ConfigSpace/configuration.py#L208-L236

I'd say the best way is actually the following then:

```python
def __contains__(self, key):
    try:
        self.__getitem__()
        return True
    except KeyError:
        return False
```

**Comment by eddiebergman** [comment]
Released in `1.1.4`
